app/controllers/api/v2/hosts_controller.rb

```diff
@@ -17,6 +17,18 @@ class HostsController < V2::BaseController
 
       add_smart_proxy_filters :facts, :features => proc { Foreman::Plugin.fact_importer_registry.fact_features }
 
+      VIRT_KEYS = {
+        "ec2" => [:availability_zone, :public_ip_address, :dns_name, :private_ip_address, :private_dns_name,
+                  :kernel_id, :state, :created_at, :root_device_type, :image_id, :flavor_id, :groups],
+        "gce" => [:state, :pretty_machine_type, :zone_name, :creation_timestamp, :private_ip_address,
```

Comment:
Note that in https://github.com/theforeman/foreman/pull/9272 gce will be removed (to be replaced with a `foreman_google` plugin. Is this somehow pluggable?

Perhaps this array should be defined in the compute resource provider.